### ISSUE TITLE
Style/NegatedIfElseCondition bugs with specific conditions

### ISSUE THREAD

**Issue by edouard-per-angusta** [rubocop/rubocop#9731]
The Style/NegatedIfElseCondition cop bugs with very specific conditions with error `Parser::Source::TreeRewriter detected clobbering`.

```
1 error occurred:
An error occurred while Style/NegatedIfElseCondition cop was inspecting ...
```
--------

## Steps to reproduce the problem

Create a file with the code below and run rubocop on it:

```ruby
if !a_condition
  # comment
  flag = false
  do_a
end

if !another_condition
  # comment
  flag = false
  do_b
else
  do_c
end
```

This seems very specific since it works if you remove the else block, if you remove the comments...

## RuboCop version

```
$ bundle exec rubocop -V
1.12.1 (using Parser 3.0.1.0, rubocop-ast 1.4.1, running on ruby 2.6.7 x86_64-darwin20)
  - rubocop-performance 1.9.2
  - rubocop-rails 2.9.1
  - rubocop-rspec 2.2.0
```

**Comment by dvandersluis** [comment]
This is an interesting bug indeed. I found an actual issue in `Style/NegatedIfElseCondition` that I have a fix for, where branches aren't being swapped properly when they are only a single statement.

However, there's also a second issue with how the AST is associated with comments that is being tripped for this specific example because there are two identical nodes (the `flag = false` lines) that each have an attached comment. The code to swap the if-else branches gets a range for each branch that includes comments, but because of rubocop/rubocop-ast#179, in your specific example it considers this entire part of the code to be the "if branch":

```ruby
if !a_condition
  # comment                     # <== start
  flag = false
  do_a
end

if !another_condition
  # comment
  flag = false
  do_b                         # <== end
else
  do_c
end
```
